community/values/src/main/java/org/neo4j/values/storable/PointValue.java

```diff
@@ -280,9 +280,32 @@ public static PointValue fromMap( MapValue map )
     private static Pattern quotesPattern = Pattern.compile( "^[\"']|[\"']$" );
 
     public static PointValue parse( CharSequence text )
+    {
+        return PointValue.parse( text, null );
+    }
+
+    public static PointValue parse( CharSequence text, AnyValue[] fieldsFromHeader )
+    {
+        AnyValue[] fieldsFromData = parseIntoArray( text );
+        if ( fieldsFromHeader != null )
+        {
+            //It is given that fieldsFromData.length == fieldsFromHeader.length because parseIntoArray produces fixed length arrays
+            // Merge InputFields: Data fields override header fields
```

Comment:
That should have a JavaDoc (e.g. `fieldsFromHeader must be a value obtained from parseIntoArray` ) and an actual assert statement would also be nice.

Reply:
Or with if we do the change to `defaultFields` we can just do the for loop from 0 to min(fieldsFromData.length, defaultFields.length).